packages/core/src/did/Did.utils.ts

```diff
@@ -143,7 +141,7 @@ export function verifyDidDocumentSignature(
   if (identifier !== id) {
     throw SDKErrors.ERROR_DID_IDENTIFIER_MISMATCH(identifier, id)
   }
-  const unsignedDidDocument = { ...didDocument }
+  const unsignedDidDocument = { ...didDocument, signature: undefined }
```

Comment:
Why is that necessary, when we remove the signature right after?

Reply:
The linter complained about the deletion.

Reply:
I am happy to remove this as this is independent of the PR.

Reply:
proposal: instead of deletion below, why don't we do 
```typescript
const { signature, ...unsignedDidDocument } = didDocument
```

Reply:
Not sure whether this is fine because then the `signature` key still exists.

Reply:
it does not in unsignedDidDocument, but of course we then have a variable in scope with that name - not sure how to improve this (the linter will complain)

Reply:
Okay so I will change it to this, which won't have a linter complain:
```ts
  const { signature, ...unsignedDidDocument } = didDocument
  return Crypto.verify(
    Crypto.hashObjectAsStr(unsignedDidDocument),
    signature,
    getAddressFromIdentifier(identifier)
  )
```

Reply:
Fixed in 692eca0

Reply:
oooh I love it! didn't see we need the signature anyway, that's perfect.